app/assets/javascripts/admin/analytics-modules/ga4-finder-setup.js

```diff
@@ -0,0 +1,124 @@
+'use strict'
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {}
+window.GOVUK.analyticsGa4.analyticsModules =
+  window.GOVUK.analyticsGa4.analyticsModules || {}
+;(function (Modules) {
+  Modules.analyticsModules = Modules.analyticsModules || {}
+
+  const excludedTypes = ['search', 'hidden']
+
+  Modules.analyticsModules.Ga4FinderSetup = {
+    init: function () {
+      const finders = Array.from(
+        document.querySelectorAll("[data-module~='ga4-finder-tracker']")
+      )
+
+      if (finders) {
+        finders.forEach((finder) => {
+          Array.from(finder.querySelectorAll('input, select')).forEach((el) => {
+            const changeCategory = el.getAttribute('data-ga4-change-category')
+            const isDateInput = el.closest('.govuk-date-input')
+
+            if (excludedTypes.find((type) => type === el.type)) {
+              if (!changeCategory && !isDateInput) return
+            }
+
+            if (!changeCategory) {
+              let changeCategoryType
+
+              const isDateInput = el.closest('.govuk-date-input')
+              const isSelect = el.tagName === 'SELECT'
+
+              if (isSelect) {
+                changeCategoryType = 'select'
+              }
+
+              if (isDateInput) {
+                changeCategoryType = 'date'
+              }
+
+              changeCategoryType = changeCategoryType || el.type
+
+              if (el.hasAttribute('multiple')) {
+                changeCategoryType = `${changeCategoryType}-multiple`
+              }
+
+              el.setAttribute(
+                'data-ga4-change-category',
+                `update-filter ${changeCategoryType}`
+              )
```

Comment:
It's quite difficult to follow the procedural style code above - is there any way it could be broken down into well named methods that are called out to instead? E.g.

```javascript
init: function () {
  finders().forEach(finder => {
    inputs(finder).forEach(input => {
      changeCategoryType = figureOutTheChangeType(input)
      if (changeCategoryType) {
        input.setAttribute(
          'data-ga4-change-category',
        `update-filter ${changeCategoryType}`
        )
    })
  })
```